src/main/java/seedu/duke/controllers/ModulePlannerController.java

```diff
@@ -177,6 +166,34 @@ public void start() {
                     }
                     break;
                 }
+                case "required": {
+                    getRequiredModules(student.getMajor());
+                    break;
+                }
+                case "info": {
+                    view.displayMessage("info");
+                    try {
+                        Api.infoCommands(words[1], userInput);
+                    } catch (UnknownCommandException e) {
+                        ErrorHandler.invalidCommandforInfoCommand();
+                    }
+                    break;
+                }
+                case "search": {
+                    view.displayMessage("search");
+                    if (!Parser.isValidKeywordInput(userInput)) {
```

Comment:
(should change) can you functions not throw the inside this function? so catch the error inside the function itself(can text me if u dont know what I mean regarding this)

(good to have)can all these code be attracted into another function?